homeassistant/components/onvif/camera.py

```diff
@@ -516,7 +516,8 @@ def fetch():
                 """Read image from a URL."""
                 try:
                     response = requests.get(self._snapshot, timeout=5, auth=auth)
-                    return response.content
+                    if response.status_code < 300:
+                        return response.content
```

Comment:
The fetch function has inconsistent return values. We should `return None` explicitly in the two other exits of this function.